crates/core_arch/src/x86/avx.rs

```diff
@@ -792,16 +792,12 @@ pub const _CMP_TRUE_US: i32 = 0x1f;
 /// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_cmp_pd)
 #[inline]
 #[target_feature(enable = "avx,sse2")]
-#[cfg_attr(test, assert_instr(vcmpeqpd, imm8 = 0))] // TODO Validate vcmppd
-#[rustc_args_required_const(2)]
+#[cfg_attr(test, assert_instr(vcmpeqpd, IMM8 = 0))] // TODO Validate vcmppd
+#[rustc_legacy_const_generics(2)]
 #[stable(feature = "simd_x86", since = "1.27.0")]
-pub unsafe fn _mm_cmp_pd(a: __m128d, b: __m128d, imm8: i32) -> __m128d {
-    macro_rules! call {
-        ($imm8:expr) => {
-            vcmppd(a, b, $imm8)
-        };
-    }
-    constify_imm6!(imm8, call)
+pub unsafe fn _mm_cmp_pd<const IMM8: i32>(a: __m128d, b: __m128d) -> __m128d {
```

Comment:
The name `IMM8` is misleading here since the immediate isn't actually 8 bits.

Reply:
I can change that. I thought the old `imm8` name was going for consistency with the Intel intrinsics guide.

Reply:
Oh hmm, I didn't notice that the intrinsics guide actually used `imm8`. That's just misleading...

I'll leave it up to your discretion as to whether to rename the argument or not.

Reply:
Yeah it seems it does so more often than not, misleading indeed.

We do deviate in the naming here and there already, so it's probably ok for us to be more precise in the naming and have `IMM5` here ? We could automate renaming everything to match the guide if we needed to some day. 

I'll do a quick pass for the intrinsics in this PR, and we'll see if everyone likes it :)

Reply:
I've pushed 210cf93 to address this, for these new converted intrinsincs:
- whenever Intel has an `imm8` of width different from 8, I went with the width, like `IMM5`, so that we're more precise
- whenever they have another name not mentioning the width, for example we're already using `INDEX` in a couple places, I went with that (even though I wish we could still mention the width)

what do you think ?

Reply:
Looks great!